prometheus/promhttp/http.go

```diff
@@ -381,6 +391,34 @@ type HandlerOpts struct {
 	ProcessStartTime time.Time
 }
 
+type OpenMetricsOptions struct {
+	// If true, the experimental OpenMetrics encoding is added to the
+	// possible options during content negotiation. Note that Prometheus
+	// 2.5.0+ will negotiate OpenMetrics as first priority. OpenMetrics is
+	// the only way to transmit exemplars. However, the move to OpenMetrics
+	// is not completely transparent. Most notably, the values of "quantile"
+	// labels of Summaries and "le" labels of Histograms are formatted with
+	// a trailing ".0" if they would otherwise look like integer numbers
+	// (which changes the identity of the resulting series on the Prometheus
+	// server).
+	EnableOpenMetrics bool
+	// If 'EnableOpenMetrics' is true, 'EnableOpenMetricsCreatedMetrics' allows
+	// to add extra '_created' lines for counters, histograms and summaries,
+	// as defined in the OpenMetrics specification (see
+	// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter-1).
+	// Created timestamps are used to improve the accuracy of reset detection,
+	// but be aware that it also increases the size of the payload.
+	//
+	// Prometheus introduced the feature flag 'created-timestamp-zero-ingestion'
+	// in version 2.50.0, but with support limited to the Prometheus protobuf
+	// format. Starting in Prometheus XXXX, the feature flag will be extended
+	// to the OpenMetrics text format. If using Prometheus XXXX or later, it
+	// is recommended to enable the feature flag in Prometheus, otherwise enabling
+	// _created lines will result in increased cardinality and no improvements
+	// in reset detection.
```

Comment:
```suggestion
	// EnableCreatedSeries specifies if this handler should add, extra, synthetic
	// '_created' series for counters, histograms and summaries, as defined in
	// the OpenMetrics specification (see
	// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter-1).
	// 
	// Created timestamps are used to improve the accuracy of reset detection,
	// but the way it's designed in OpenMetrics 1.0 it also dramatically increases cardinality
	// if the scraper does not handle those metrics correctly (converting to created timestamp
	// instead of leaving those series as-is). New OpenMetrics versions might improve
	// this situation.
	//
	// Prometheus introduced the feature flag 'created-timestamp-zero-ingestion'
	// in version 2.50.0, but only for the Prometheus protobuf format. Starting in 
	// future Prometheus version, the feature flag will be extended to the OpenMetrics
	// text format, thus safe to be enabled to improve accuracy of counters in Prometheus.
```